### PR TITLE
Make denomination configurable

### PR DESCRIPTION
**Pull request by vitorenesduarte**
## Description

This PR allows the denomination to be configurable.
It also removes what looks like an unnecessary conversion from/to JSON.

______

For contributor use:

- [ ] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: relayer-cli/src/commands/tx/transfer.rs [hunk 1]
```diff
@@ -30,12 +30,18 @@ pub struct TxRawSendPacketCmd {
...
+        free,
+        help = "amount of coins (samoleans, by default) to send (e.g. `100000`)"
+    )]
     amount: u64,
 
     #[options(free, help = "timeout in number of blocks since current")]
     height_offset: u64,
 
+    #[options(help = "denomination of the coins to send", short = "d")]
+    denom: Option<String>,
```
Comment (romac) [inline relayer-cli/src/commands/tx/transfer.rs:43]: Can we introduce an `enum` for the supported denominations, even if it's only samoleans at the moment?

Comment (vitorenesduarte) [reply relayer-cli/src/commands/tx/transfer.rs:43]: My understanding is that denominations are dynamic and can take any value at runtime. For example, after the steps in https://github.com/informalsystems/ibc-rs/blob/master/relayer-cli/relayer_operation_instructions.md, to send the coins back to `ibc-0`, we would need to use something like `ibc/27A6394C3F9FF9C9DCF5DFFADF9BB5FE9A37C7E92B006199894CF1824DF9AC7C` as the denominator. Such denominator can be found with this query: https://github.com/informalsystems/ibc-rs/blob/master/one-chain#L137

Comment (vitorenesduarte) [reply relayer-cli/src/commands/tx/transfer.rs:43]: Interestingly, I tried to do that and the coins disappeared (both balances are 0), so I'm likely doing something wrong. If someone wants to give it a try, at [this commit](https://github.com/vitorenesduarte/ibc-rs/commit/cb24b8826687d2885bc34bd130093e4c3607647d), the following command should do it: `echo y | dev-env loop_config.toml ibc-0 ibc-1 && instructions.sh`

Comment (ancazamfir) [reply relayer-cli/src/commands/tx/transfer.rs:43]: > My understanding is that denominations are dynamic and can take any value at runtime

This is correct.

> Interestingly, I tried to do that and the coins disappeared (both balances are 0),..

I tried to transfer some `samoleans` from `ibc-0` to `ibc-1` and then `ibc/27A6394C3F9FF9C9DCF5DFFADF9BB5FE9A37C7E92B006199894CF1824DF9AC7C` tokens back to `ibc-0` from `ibc-1`. Things seem to work. Note that these `ibc/....` tokens sent to original chain show back into the `samoleans` denom balance for `ibc-0`.
Also make sure you complete the handshake (i.e. do the `tx raw packet-recv...` and `tx raw packet-ack...` CLIs)

I will try more tests later. But I think this is good to go.

**Review by ancazamfir — COMMENTED** [review]
Looks good, thanks Vitor! Could you fix clippy errors and merge master.

File: relayer-cli/src/commands/tx/transfer.rs (hunk 1 above)
Comment (ancazamfir) [reply relayer-cli/src/commands/tx/transfer.rs:43]: Just looked now at `instructions.sh` in the other commit...in lines 65-67 you have to flip `ibc-0` and `ibc-1`. It should look like this:

```
DENOM="ibc/27A6394C3F9FF9C9DCF5DFFADF9BB5FE9A37C7E92B006199894CF1824DF9AC7C"
rrly tx raw packet-send ibc-1 ibc-0 transfer channel-0 $AMOUNT $TIMEOUT -n 1 -d $DENOM
rrly tx raw packet-recv ibc-1 ibc-0 07-tendermint-0 07-tendermint-0 transfer transfer channel-0 channel-0
rrly tx raw packet-ack ibc-1 ibc-0 07-tendermint-0 07-tendermint-0 transfer transfer channel-0 channel-0
rrly query packet unreceived-packets ibc-0 ibc-1 transfer channel-0
rrly query packet unreceived-acks ibc-1 ibc-0 transfer channel-0
```

**Comment by ancazamfir** [comment]
Could you please also update the instructions [here](https://github.com/informalsystems/ibc-rs/blob/master/relayer-cli/relayer_operation_instructions.md) with a new section on how to query balances and transfer tokens back and forth?
We should also update CHANGELOG.md, ideally there should be an issue opened for this.

File: relayer-cli/src/commands/tx/transfer.rs (hunk 1 above)
Comment (vitorenesduarte) [reply relayer-cli/src/commands/tx/transfer.rs:43]: Thanks for pointing it out. It's working as expected now.

**Review by ancazamfir — APPROVED** [review]
LGTM!!! Thanks!

**Pull request merged**